### DISCUSSION TITLE
Add Speech Recognition to search module to improve accessibility

### DISCUSSION THREAD

**Discussion by Sulpher** [joomla/joomla-cms#32497] [2. Cold Features]
Voice command recognition to any website makes it more accessible and user-friendly. By adding voice search HTML code to a website, users are allowed to make their search requests instantly without the need of having any experience of touch typing. And it is a good signal to search engines that potentially can positively impact site ranking.

Why not implementing such a solution to Joomla search module?

Here are some proofs:
https://codepen.io/Coding_Journey/pen/ewrWyG
https://www.labnol.org/software/add-speech-recognition-to-website/19989/

**Comment by brianteeman** [comment]
I've done a lot of work on text-to-speech and speech-to-text. It is incredibly easy to implement if you want it although I doubt its usefulness. I have never heard of it having any site ranking benefits. Could you explain more on that please

**Comment by Sulpher** [comment]
> I've done a lot of work on text-to-speech and speech-to-text. It is incredibly easy to implement if you want it although I doubt its usefulness

Do you mean search module in J4?

> Could you explain more on that please
I've read about this technique in Russian, however, some good articles just been recently found: 
https://www.semrush.com/blog/voice-search-optimization-7-seo-strategies-to-rank-better/
https://www.singlegrain.com/seo/effective-seo-techniques-that-work/

Voice Search is one of the factors which indirectly affects SEO like speed optimization, mobile-friendly template and etc.

**Reply by brianteeman** [reply]
Both of those articles are referring to optimising your content for off-site voice search which is not the same as adding speech recognition to your own site.

**Comment by Sulpher** [comment]
But the general idea is voice search feature is useful for visitors and indirectly impacts on SEO.

Have you already implemented it in J4 search module or you made it on your own websites?

**Reply by brianteeman** [reply]
As explained before and in those articles - adding voice search to your own site will have ZERO benefit for SEO

As for it being useful for visitors. 
If they are using a mobile phone etc then there is nothing you need to do to your site. It already has the ability to use your voice to "input" search terms in the search module.
On the desktop if a user found it useful they could install one of many browser plugins.

Those articles are referring to voice search similar to Alexa, Siri, OK Google. And yes there are ways to write your content so that it is more discoverable to typical voice searches
